components/brave_vpn/common/wireguard/wireguard_utils.cc

```diff
@@ -82,6 +88,102 @@ WireguardKeyPair GenerateNewX25519Keypair() {
       EncodeBase64(std::vector<uint8_t>(privkey, privkey + 32)));
 }
 
+bool ValidateKey(const wchar_t* key,
+                 std::string* output,
+                 const std::string field_name) {
+  if (!base::WideToUTF8(key, wcslen(key), output)) {
+    VLOG(1) << "failed decoding " << field_name;
+    return false;
+  }
+
+  base::TrimWhitespaceASCII(*output, base::TRIM_ALL, output);
+
+  if ((*output).length() == 0) {
+    VLOG(1) << field_name << " does not have a value";
+    return false;
+  }
+
+  if (!re2::RE2::FullMatch(*output, R"(^[-A-Za-z0-9+\/=]+$)")) {
+    VLOG(1) << field_name << " contains invalid characters";
+    return false;
+  }
+
+  std::string decoded_config;
+  if (!base::Base64Decode(*output, &decoded_config) || decoded_config.empty()) {
+    VLOG(1) << field_name << " is not base64 encoded";
+    return false;
+  }
+
+  if (decoded_config.length() != 32) {
+    VLOG(1) << field_name << " is not the correct length";
+    return false;
+  }
+
+  return true;
+}
+
+bool ValidateAddress(const wchar_t* address, std::string* output) {
```

Comment:
use absl::optional instead of out params

Reply:
Fixed in latest commit